daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/Converter.scala

```diff
@@ -675,7 +675,8 @@ object Converter {
         } catch {
           case e: Exception => Left(s"LF conversion failed: ${e.toString}")
         }
-      valueTranslator = new preprocessing.ValueTranslator(compiledPackages.interface)
+      valueTranslator =
+        new preprocessing.ValueTranslator(compiledPackages.interface, requiredCidSuffix = false)
```

Comment:
It will not blow up. `requiredCidSuffix = false` corresond to your "don’t care" modes, 

There is currently no mode to "enforcing no suffixes". I am not sure it is very useful.